src/devices/Board/CustomBoard.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.Text;
+
+namespace Iot.Device.Board
+{
+    /// <summary>
+    /// A board that can be customized with user-specified drivers.
+    /// This should only be used if the drivers can't be auto-detected properly.
+    /// </summary>
+    public class CustomBoard : GenericBoard
+    {
+        private readonly GpioDriver _gpioDriver;
+        private readonly Func<int, I2cBus> _i2CBusCreator;
+        private readonly Func<SpiConnectionSettings, SpiDevice> _spiDeviceCreator;
```

Comment:
Can a board have multiple SPI devices? In this configuration, I'm not sure. Same question for PWM.
Question is then: shall it be a collection for both?
And tricky part: a SPI device, can be the same as another one with a little difference on the chip select. Which can make all this a bit more complicated

Reply:
Yes, a board can have multiple SPI devices. There's a special feature called "Pin sharing" that allows the board to specify that for some features (like SPI, or previously I2C), the same set of pins can be opened multiple times. In other words: It is possible to open the same pins multiple times in SPI mode, and reference counting is performed to ensure all devices are closed before a pin is used for something else.